ports/protobuf/portfile.cmake

```
@@ -28,24 +28,35 @@ endif()
 
 if (VCPKG_DOWNLOAD_MODE)
     # download PKGCONFIG in download mode which is used in `vcpkg_fixup_pkgconfig()` at the end of this script.
-    # download it here because `vcpkg_configure_cmake()` halts execution in download mode when running configure process.
+    # download it here because `vcpkg_cmake_configure()` halts execution in download mode when running configure process.
     vcpkg_find_acquire_program(PKGCONFIG)
 endif()
 
-vcpkg_configure_cmake(
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static") 
+    set(protobuf_VISIBILITY_FLAGS 
+        -DCMAKE_POLICY_DEFAULT_CMP0063=NEW
+        -DCMAKE_C_VISIBILITY_PRESET=hidden
+        -DCMAKE_CXX_VISIBILITY_PRESET=hidden
+        -DCMAKE_VISIBILITY_INLINES_HIDDEN=TRUE
+    )
+else() 
+    set(protobuf_VISIBILITY_FLAGS )
```

```suggestion
    set(protobuf_VISIBILITY_FLAGS "")
```

we don't want to unset `protobuf_VISIBILITY_FLAGS`

---

@JackBoosY you need to `set(blah "")` in order to give it a value; otherwise, it's just equivalent to `unset(blah)`